Tests/UnitTests/Core/EventData/TransformFreeToBoundTests.cpp

```diff
@@ -42,6 +42,9 @@ BOOST_DATA_TEST_CASE(
   // transform reference position
   Vector3 pos = surface->localToGlobal(geoCtx, loc, dir);
 
+  // @TODO: Add a test for the failing case, i.e. the parameters are not on
+  // surface.
+
```

Comment:
Would you mind adding this, these TODO's usually hang around for a long time.

Reply:
Ha, that's why I put the TODO in, so I don't forget 😅

Reply:
Should be added.